platform/features/table/src/controllers/TableController.js

```diff
@@ -0,0 +1,214 @@
+            this.changeListeners = [];
+
+            $scope.rows = [];
+
+            // Subscribe to telemetry when a domain object becomes available
+            this.$scope.$watch('domainObject', function(domainObject){
+                if (!domainObject)
+                    return;
+
+                self.subscribe();
+                self.registerChangeListeners();
+            });
+
+            // Unsubscribe when the plot is destroyed
+            this.$scope.$on("$destroy", this.destroy.bind(this));
+        }
+
+        TableController.prototype.registerChangeListeners = function() {
+            //Defer registration of change listeners until domain object is
+            // available in order to avoid race conditions
+
+            this.changeListeners.forEach(function (listener) {
+                return listener && listener();
+            });
+            this.changeListeners = [];
+            // When composition changes, re-subscribe to the various
+            // telemetry subscriptions
+            this.changeListeners.push(this.$scope.$watchCollection('domainObject.getModel().composition', this.subscribe.bind(this)));
+
+            //Change of bounds in time conductor
+            this.changeListeners.push(this.$scope.$on('telemetry:display:bounds', this.subscribe.bind(this)));
+
+        };
+
+        /**
+         * Release the current subscription (called when scope is destroyed)
+         */
+        TableController.prototype.destroy = function () {
+            if (this.handle) {
+                this.handle.unsubscribe();
+                this.handle = undefined;
+            }
+        };
+
+        /**
+         Create a new subscription. This is called when
+         */
+        TableController.prototype.subscribe = function() {
+            var self = this;
+
+            /*if (this.pending){
+                return;
+            }*/
+            //this.pending = true;
+
+            if (this.handle) {
+                this.handle.unsubscribe();
+            }
+
+            this.$scope.rows = [];
+
+            //Noop because not supporting realtime data right now
+            function noop(){
+                //self.pending = false;
+            }
+
+            this.handle = this.$scope.domainObject && this.telemetryHandler.handle(
+                    this.$scope.domainObject,
+                    noop,
+                    true // Lossless
+                );
+
+            this.handle.request({}, this.addHistoricalData.bind(this));
+
+            this.setup();
+        };
+
+        /**
+         * Add any historical data available
+         */
+        TableController.prototype.addHistoricalData = function(domainObject, series) {
+            var i;
+            //this.pending = false;
+            for (i=0; i < series.getPointCount(); i++) {
+                this.updateRows(domainObject, this.handle.makeDatum(domainObject, series, i));
```

Comment:
> Additionally, it seems wasteful to re-evaluate limits for each cell in a row, when the logic is essentially the same. Can we assign the CSS class to the entire row and not the individual cells?

The reasoning behind evaluating for each __range__ cell (doesn't do it for domains) is that with multiple range values, only one of them may be in limit violation, and it's useful to highlight which one it is.